platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -659,12 +647,12 @@ private ResolverResults resolveGraphIfRequired() {
         assertIsResolvable();
         maybeEmitResolutionDeprecation();
 
-        ResolveState currentState = currentResolveState.get();
-        if (currentState.getState() == GRAPH_RESOLVED) {
-            return currentState.getCachedResolverResults();
+        Optional<ResolverResults> currentState = currentResolveState.get();
+        if (currentState.map(ResolverResults::isFullyResolved).orElse(false)) {
```

Comment:
💭 We're doing this calculation in multiple places and it's kind of dense.  I would consider moving this behind a method:  
```
private boolean isFullyResolved(Optional<ResolverResults> results) { 
    return results.map(ResolverResults::isFullyResolved()).orElse(false)
}
```
And then this becomes:
```
if (isFullyResolved(currentState)) {
```